### ISSUE TITLE
acc: generating duplicates on missed / failed calls

### ISSUE THREAD

**Issue by jchavanton** [kamailio/kamailio#1673]
### Description
I found confusion about using the integer value of the flags when checking them.
I think this as been creating little confusing problems for a long time (not 100% sure about the exact impact).
I believe users where finding work around solutions.

acc_run_engines (integrer value of the flag)
```
  if((type==1) && (msg->flags&(e->missed_flag))) {
```
and is_eng_acc_on (integer value of the flag)
```
 if(msg->flags & e->missed_flag) {
```
and (integer values if the flag)
```
flags_to_reset |= log_missed_flag
```

However when we reset it we use the (bit position value)
```
reset_acc_flag( req, flags_to_reset ); 
#define reset_acc_flag(_rq,_flag)   (resetflag((_rq), (_flag))) 
```

Everywhere else in the application (I have seen)
We use the helper fuctions like `isflagset`, `resetflag`, etc. and use integer representing the bit position
This is why I am proposing a patch to reuse this pattern.

**Comment by jchavanton** [comment]
just a little update, I this should be handled properly since the  flags should be kept even if modified by a transaction callback, good now I need to find out why this is not taking place.
```
/* if failure handler changed flag, update transaction context */
shmem_msg->flags = faked_req->flags;
```

**Comment by jchavanton** [comment]
I found consufion about using the integer value of the flag when checking it.

acc_run_engines
```
  if((type==1) && (msg->flags&(e->missed_flag))) {
```
and 
is_eng_acc_on 
```
 if(msg->flags & e->missed_flag) {
```

However when we reset it we use the bit position value
```
reset_acc_flag( req, flags_to_reset ); 
#define reset_acc_flag(_rq,_flag)   (resetflag((_rq), (_flag))) 
```

Looking at this further

**Comment by henningw** [comment]
Commenting about the third topic directly in this issue. I don't understand this remark:
"However when we reset it we use the (bit position value)"
```
reset_acc_flag( req, flags_to_reset ); 
#define reset_acc_flag(_rq,_flag)   (resetflag((_rq), (_flag))) 
```
The function resetflag is defined in a similar logic as setflag: struct sip_msg* msg, flag_t flag.
And flag_t is defined (as you correctly identified) as unsigned int.

**Comment by henningw** [comment]
Should be fixed by pull request #1674
